packages/core/src/sync-realtime/index.ts

```diff
@@ -493,7 +493,7 @@ export const createRealtimeSync = (
       for (const log of logs) {
         if (log.blockHash !== block.hash) {
           throw new Error(
-            "Detected invalid eth_getLogs response. `log.blockHash` does not match requested block hash.",
+            `Detected inconsistent eth_getLogs response. 'log.blockHash' ${log.blockHash} does not match 'block.hash' ${block.hash}`,
```

Comment:
In this case, I think it's safe to say that this request is "invalid" because the eth_getLogs is called with `blockHash`, so it's expected the result matches